### DISCUSSION TITLE
QSL management

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#1723] [Ideas / Feature Requests]
I've been using cloudlog since I got my license (about half a year). Everything looks good, but I have some suggestions:

1. different colors (can be user defined) for different types of QSL received (paper, electronic, etc.).
2. additional options for scheduling QSLs (bureau, direct) with different color markers (can be user defined). Right now I use "requested" and "invalid" (orange and white markers). But one color for different statuses leads to confusion.
3. the display of pictures received QSL not only for eQSL

**Comment by AndreasK79** [comment]
1. Where do you mean that different colors should be used? We already have different field for QSL/eQSL/LoTW:
![image](https://user-images.githubusercontent.com/6977712/198258370-001d170c-1624-4f45-80ee-2ccd04563c4d.png)
3. You mean qsl management? That is implemented. Click on a callsign on the dashboard or in the logbook. Then go to QSL Management:
![image](https://user-images.githubusercontent.com/6977712/198258793-9b748a2e-c0cf-4f1f-b428-e92083e04de3.png)
![image](https://user-images.githubusercontent.com/6977712/198259122-f139ad28-938a-44b1-b52f-ceba00816248.png)

Uploaded QSLs can be seen under QSL and View QSL.

**Reply by phl0** [reply]
> 1. Where do you mean that different colors should be used? We already have different field for QSL/eQSL/LoTW:

And even different colors for status like requested or ignore.

![Screenshot from 2022-10-27 12-24-26](https://user-images.githubusercontent.com/7112907/198260456-cfadb123-cc0e-4cd2-bbd2-1917bfc54609.png)

**Reply by R1BLH** [reply]
Sorry for my English. :)

> > 1. Where do you mean that different colors should be used? We already have different field for QSL/eQSL/LoTW:

I mean additional statuses (and markers with additional colours) in the QSL column: 

![2022-10-27_17-12-13](https://user-images.githubusercontent.com/16667173/198309198-893a0bb9-2a7c-46df-b4e5-afcec782b7b2.png)
Screeenshot 1

In the column "sent QSL" . 
See the screenshot 1 (white arrow).
Additional statuses would be: "wil be sent via bureau" and "will be sent direct". 
This statuses would be nice set via context menu. Only two options now awailable therе: "mark QSL sent (via bureau)" and "mark QSL sent (direct)". See screenshot 2 (yellow arrow). 

![2022-10-27_17-23-26](https://user-images.githubusercontent.com/16667173/198314542-9b17bac5-d2d5-493e-8a90-017b691997da.png)
Screenshot 2

It is more simple (for me) to fill batch of paper QSL, when records are already marked,



In the column "received QSL" . 
See the screenshot 1  (yellow arrow).
Now QSL marked as received with green marker rather QSL was electrronical or paper. 
It would be nice to have two marker colours for electronical and paper QSL.

Also if you click on marker of the received eQSL you would see image of eQSL. It would be good to see image of received paper/electronical QSL on click on the corresponding received QSL marker (yellow arrow on the screenshot).

**Reply by R1BLH** [reply]
> > 1. Where do you mean that different colors should be used? We already have different field for QSL/eQSL/LoTW:
> 
> And even different colors for status like requested or ignore.
> 
> ![Screenshot from 2022-10-27 12-24-26](https://user-images.githubusercontent.com/7112907/198260456-cfadb123-cc0e-4cd2-bbd2-1917bfc54609.png)

Now I use the statuses "requested" and "ignore" to mark the QSL "via bureau" and "direct". As a result, there is confusion with the valid marks "requested" and "ignore".

**Comment by R1BLH** [comment]
@AndreasK79 have you an idea how to add additional states for QSL markers (on which part of code should i look)?

**Reply by phl0** [reply]
You would have to write a database migration script to add separate columns per QSO and then use these to display the states. In addition to that you have to extend the QSO add/edit functions.

**Reply by R1BLH** [reply]
@phl0 
Hm... It's very unusual for me. I'm a dummy in web programming and php. :) I tryed to look into code. May be it would be simplier if instead additional columns add two additional values for column COL_QSL_SENT_VIA in table TABLE_HRD_CONTACTS_V01?
Now values are:
"D" -- direct
"B" -- bureau
"E" -- electronic
"R" -- reqursted
"I" -- ignored

I suggest to add two values:

"b" -- paper QSL to be sent via bureau (would be blue triangle, for example)
"d" -- paper QSL to be sent direct   (would be yellow triangle, for example)

So it would request to make additional functions for selection of this values.

For example, i tried to add this items to QSO menu in the logbook display:
![2022-11-05_03-19-43](https://user-images.githubusercontent.com/16667173/200092146-f2643dd6-8e6c-437a-aac1-d23e9240249b.png)

But i have no idea how to implement rest functions (except copy/paste/edit).

**Comment by AndreasK79** [comment]
@cats-shadow I just looked at the code. COL_QSL_SENT_VIA is the column to use. Mark QSL Card Requested, calls the function to update 
COL_QSLSDATE and COL_QSL_SENT. The function that receives the call (in logbook_model) paperqsl_requested($qso_id, $method), has a method parameter, but that is never set.

It looks to me that it would be enough to fix this, and change mark qsl card requested to two function. Mark qsl card requested (direct) and mark qsl card requested (bureau).

**Reply by R1BLH** [reply]
@AndreasK79 
> It looks to me that it would be enough to fix this, and change mark qsl card requested to two function. Mark qsl card requested (direct) and mark qsl card requested (bureau).

Hm. this would be great! Thank you!
